docs-web/src/main/java/com/sismics/docs/rest/resource/UserResource.java

```diff
@@ -1131,8 +1129,6 @@ public Response passwordLost(@FormParam("username") String username) {
     public Response passwordReset(
             @FormParam("key") String passwordResetKey,
             @FormParam("password") String password) {
-        authenticate();
```

Comment:
Does not need authentication?

Reply:
This can be removed but you need to put null at line 1152 to avoid NPE.

Reply:
I would put something like this in the beggining of the method:
```Java
//Get the principal if logged in user
Principal attributePrincipal = (Principal) request.getAttribute(SecurityFilter.PRINCIPAL_ATTRIBUTE);
if (attributePrincipal instanceof IPrincipal) {
    this.principal = (IPrincipal) attributePrincipal;
} else {
    //What to do here?
}
```
This way the `AuditLog` will contain the user which did the reset if possible.

I know its code duplication but `authenticate()` is just not semantically correct 🤔.
Also should we just throw a `ForbiddenClientException` if the principal is not of `IPrincipal`? Should it always be of `IPrincipal`? Would another error be more appropriate?

Reply:
This principal is only filled in SecurityFilter so there should never be any casting issue.
I agree that this authenticate() smells a bit, it's actually more of a "isAuthenticated()" or something like this.

Reply:
I think `isAuthenticated()` would smell even more because the method changes the global state which would then not be implied. I will try to think of a good solution regarding this method.